rtloader/three/three.cpp

```diff
@@ -197,6 +197,9 @@ bool Three::addPythonPath(const char *path)
 
 rtloader_gilstate_t Three::GILEnsure()
 {
+    if (_Py_IsFinalizing()) {
+        return DATADOG_AGENT_RTLOADER_GIL_FINALIZING;
```

Comment:
in this case, `GILEnsure` will not actually ensure that we can call the C python API once it returns. Do you know what happens if the GIL is finalizing in this thread and the code that called `GILEnsure` still continues to make calls to the C Python API?